sampleChangerApp/src/converter.cpp

```diff
@@ -219,8 +238,14 @@ std::string converter::get_available_in_slot(std::string slot)
     std::string res;
     std::list<std::string> positions;
     try {
-        positions = m_positions_for_each_slot.at(slot);
-    } catch (const std::out_of_range &e) {
+        std::vector<SlotPositions>::iterator it = find_in_positions(slot);
+        if (it == v_positions_for_each_slot.end())
+        {
+            throw std::out_of_range("not available in slot");
+        }
+        positions = it->positions;
+    }
+    catch (const std::out_of_range& e) {
         errlogPrintf("Slot '%s' unknown, returning all positions\n", slot.c_str());
         slot = ALL_POSITIONS_NAME;
     }
```

Comment:
`slot = ALL_POSITIONS_NAME` doesn't actually do anything as `slot` is being passed by value. I know this is not part of your changes, but we should check what behaviour we wish here as currently the slot name will be unchanged although all positions will be returned.